### basecalling
fast5 -> fasta/fastq
`guppy_basecaller -i {raw_fast5_dir} -s {dest_dir} --flowcell {flowcell_id} --kit {Kit_id} -r --calib_detect --enable_trimming true --trim_strategy rna --reverse_sequence true`

`cat {dir_to guppy output}/*.fastq > {basecalled_fastq}`

### mapping/alignment
how reads relate to reference

mapping - for sake of mapping
alignment - for e.g. splicing sake

`minimap2`
`minimap2 [options] <ref.fa>|<ref.idx> [query.fa]`
	- map/align (ref.fa read.fq)
	- can index (make files smaller&more digestable for further analysis)
	- don't need to decompress gzip's

`minimap2 -ax map-ont -L --split-prefix=tmp ref.fa read.fq | samtools view -bh | samtools sort -O bam > reads.bam`  - works best

draft:
	`minimap2 -ax map-ont -L --split-prefix=tmp {transcriptome_fasta} {basecalled_fastq} | samtools view -bh -F 2324 -q 10 | samtools sort -O bam > {aligned_reads.bam}`
		`-a` output in SAM (PAF default), `-x` preset (map-ont i.e. Nanopore vs reference mapping by default)
		`-L` write `CIGAR` string with >65535 ops to the CG tag (to combat bam design flaw)
		`transcriptome_fasta` - I'm pretty sure we have to use transcryptome ref OR `awk`/`sed` nucleotides
			should accept `.gz` & `.bz`
		`--split-prefix=tmp` bos I don't have enough memory
		`view` - sam<->bam<->cram conversion
		`b` BAM, `-h` with SAM header
		~~`-F 2324` exclude flag 2324 (exclude reads mapped on the reverse strand) ~~
		~~`-q 10` have mapping quality >= 10  (very low alignment score (MAPQ<10)?)~~
		`-O` output format



****
if "no SQ lines in header" during sam->bam:
```
minimap2 indexes 4 billion reference bases (4Gb) in a batch and map all reads against each reference batch. Given a reference longer than 4Gb, minimap2 is unable to see all the sequences and thus can't produce a correct SAM header. In this case, minimap2 doesn't output any SAM header.
There are two solutions to this issue.
-> First, you may increase option `-I` to, for example, `-I8g` to index more reference bases in a batch. This is preferred if your machine has enough memory.
-> Second, if your machines doesn't have enough memory to hold the reference index, you can use the `--split-prefix` option in a command line like:
```
`minimap2 -ax map-ont --split-prefix=tmp ref.fa reads.fq`
****




### indexing
`samtools index aligned_reads.bam` --> `aligned_reads.bam.bai`

`samtools faidx ref.fa` --> `ref.fai`
	could also use `.fa.gz` (results are the same, even if filname is `.fa.gz.fai`)

[latest picard](https://github.com/broadinstitute/picard/releases/latest) - set of tools for sam,bam,cram&vcf in java (needs java jdk)
`java -jar picard.jar CreateSequenceDictionary R=ref.fa O=ref.fa.dict`
		`R` reference could be both `.fa` or `.fa.gz`
	`ref.dict` - sequence dictionary file - for easy access to metadata later


### docker
`sudo docker build -t epi12 .`
	build `epi12` docker container with everything needed
	run in dir with Dockerfile

`sudo docker run -it -d --rm --name epivar -v "$PWD/data/":/project/ epi12 python3 /usr/local/bin/EpiNano/Epinano_Variants.py -R /project/ref.fa -b /project/reads.bam -s /usr/local/bin/EpiNano/misc/sam2tsv.jar -n 2`
	`docker run` - starts container `epi12` and name it as `epivar`
	`-it` interactive tty (open stdin on a pseudo-tty)
	`-d` detached mode (in the background) - you might not want to use that
	`--rm` remove the container when it exits (for easy cleanup sake)
	`--name epivar2` name container "epivar2"
	`-v "$PWD/data/":project/` - mount `$PWD/data/` to a directory `/project/` inside container
		mounted dir must have:
		- `ref.fa`
		- `ref.fa.fai`
		- `ref.fa.dict`
		- `reads.bam`
		- `reads.bam.bai` (indexing by the `Epinano_Variations.py` DON'T WORK)

empty dataframe for /project/reads.plus_strand.per.site.csv  ---> might be a result of mapping with `-F 2324` flag! -> Nope:<

`df = dd.read_csv ("{}/small_*freq".format(tmp_dir))` - df staje sie empty w tym momencie - KLUCZ
```python
df = dd.read_csv('my_file.csv', low_memory=False)
```

