### basecalling
fast5 -> fasta/fastq
`guppy_basecaller -i {raw_fast5_dir} -s {dest_dir} --flowcell {flowcell_id} --kit {Kit_id} -r --calib_detect --enable_trimming true --trim_strategy rna --reverse_sequence true`

`cat {dir_to guppy output}/*.fastq > {basecalled_fastq}`

### mapping/alignment
how reads relate to reference

mapping - for sake of mapping
alignment - for e.g. splicing sake

`minimap2`
`minimap2 [options] <ref.fa>|<ref.idx> [query.fa]`
	- map/align (ref.fa read.fq)
	- can index (make files smaller&more digestable for further analysis)
	- don't need to decompress gzip's

`minimap2 -ax map-ont -L {transcriptome_fasta} {basecalled_fastq} | samtools view -bh -F 2324 -q 10 | samtools sort -O bam > {aligned_reads_bam}`
	`-a` output in SAM (PAF default), `-x` preset (map-ont i.e. Nanopore vs reference mapping by default)
	`-L` write `CIGAR` string with >65535 ops to the CG tag (to combat bam design flaw)
	`transcriptome_fasta` - I'm pretty sure we have to use transcryptome ref OR `awk`/`sed` nucleotides
	`view` - sam<->bam<->cram conversion
	`b` BAM, `-h` with SAM header
	`-F 2324` exclude flag 2324 (exclude reads mapped on the reverse strand)
	`-q 10` have mapping quality >= 10  (very low alignment score (MAPQ<10)?)
	`-O` output format

****
if "no SQ lines in header" during sam->bam:
```
minimap2 indexes 4 billion reference bases (4Gb) in a batch and map all reads against each reference batch. Given a reference longer than 4Gb, minimap2 is unable to see all the sequences and thus can't produce a correct SAM header. In this case, minimap2 doesn't output any SAM header.
There are two solutions to this issue.
-> First, you may increase option `-I` to, for example, `-I8g` to index more reference bases in a batch. This is preferred if your machine has enough memory.
-> Second, if your machines doesn't have enough memory to hold the reference index, you can use the `--split-prefix` option in a command line like:
```
`minimap2 -ax map-ont --split-prefix=tmp ref.fa reads.fq`
****

`samtools index {aligned_reads_bam}`



