`minimap2 -a -L Homo_sapiens.GRCh38.dna.toplevel.fa.gz FAU81678_pass_351f87f2_0.fastq | samtools view -bh -F 2324 -q 10 | samtools sort -O bam > aligned_reads.bam`
	- do I have/should I remove *reverse strand* flag? (`-F 2324`)
	- is deleting low alignment score (`-q 10`) at this moment a good idea?




#### docker tests
`sudo docker build -t epi12 .`
	build `epi12` docker container with everything needed
	run in dir with Dockerfile

`sudo docker run -it -d --rm --name epivar -v "$PWD/data/":/project/ epi12 python3 /usr/local/bin/EpiNano/Epinano_Variants.py -R /project/ref.fa -b /project/reads.bam -s /usr/local/bin/EpiNano/misc/sam2tsv.jar -n 2`
	`docker run` - starts container `epi12` and name it as `epivar`
	`-it` interactive tty (open stdin on a pseudo-tty)
	`-d` detached mode (in the background) - you might not want to use that
	`--rm` remove the container when it exits (for easy cleanup sake)
	`--name epivar2` name container "epivar2"
	`-v "$PWD/data/":project/` - mount `$PWD/data/` to a directory `/project/` inside container

`sudo docker run -it --rm --name epivar2 -v /home/arco/Workspace/:/Workspace epi12 python3 /Workspace/EpiNano/Epinano_Variants.dev.py -r /Workspace/align/data/ref.fa -b /Workspace/align/data/reads_noF_noq.bam -c 3`




`python3 /Workspace/EpiNano/Epinano_Variants.dev.py -r /Workspace/align/data/ref.fa -b /Workspace/align/data/reads_noF_noq.bam -c 3`
```python
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/usr/local/bin/Python-3.6.3/Lib/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/bin/Python-3.6.3/Lib/multiprocessing/pool.py", line 47, in starmapstar
    return list(itertools.starmap(args[0], args[1]))
  File "/Workspace/EpiNano/Epinano_Variants.dev.py", line 74, in bam_to_var
    out = open (outfn, 'w')
FileNotFoundError: [Errno 2] No such file or directory: '/Workspace/align/data/reads_noF_noq.tmp//Workspace/align/data/reads_noF_noq.fwd.4.per.site.csv'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Workspace/EpiNano/Epinano_Variants.dev.py", line 153, in <module>
    main ()
  File "/Workspace/EpiNano/Epinano_Variants.dev.py", line 143, in main
    multi_processes_bam2var(fafn, ref_names, bam, strands[i], ncpus,  outdir)
  File "/Workspace/EpiNano/Epinano_Variants.dev.py", line 109, in multi_processes_bam2var
    files = pool.starmap (bam_to_var,conditions)
  File "/usr/local/bin/Python-3.6.3/Lib/multiprocessing/pool.py", line 274, in starmap
    return self._map_async(func, iterable, starmapstar, chunksize).get()
  File "/usr/local/bin/Python-3.6.3/Lib/multiprocessing/pool.py", line 644, in get
    raise self._value
FileNotFoundError: [Errno 2] No such file or directory: '/Workspace/align/data/reads_noF_noq.tmp//Workspace/align/data/reads_noF_noq.fwd.4.per.site.csv'
```


# **The input BAM file may not have any reads mapped to the reference sequence(s) specified.**



(not docker stuff) - Annoying:
```
UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - pysam -> python[version='>=3.7,<3.8.0a0']

Your python: python=3.10

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to. Note that conda will not
change your python version to a different minor version unless you explicitly specify
that.

The following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.37=0
  - feature:|@/linux-64::__glibc==2.37=0

Your installed version is: 2.37
```
I can't install `pysam` due to this (and I should have `python 3.6.7`)

while trying to pip install - legacy-install-failure of `h5py` and metadana-generation-failed of `pandas` (installable with `conda`)